monkey/infection_monkey/island_api_client/i_island_api_client.py

```diff
@@ -108,38 +109,37 @@ def register_agent(self, agent_registration_data: AgentRegistrationData):
         """
 
     @abstractmethod
-    def should_agent_stop(self, agent_id: str) -> bool:
+    def get_config(self) -> AgentConfiguration:
         """
-        Check with the island to see if the agent should stop
+        Get agent configuration from the island
 
-        :param agent_id: The agent identifier for the agent to check
         :raises IslandAPIConnectionError: If the client could not connect to the island
         :raises IslandAPIRequestError: If there was a problem with the client request
         :raises IslandAPIRequestFailedError: If the server experienced an error
         :raises IslandAPITimeoutError: If the command timed out
-        :return: True if the agent should stop, otherwise False
+        :return: Agent configuration
         """
 
     @abstractmethod
-    def get_config(self) -> AgentConfiguration:
+    def get_credentials_for_propagation(self) -> Sequence[Credentials]:
         """
-        Get agent configuration from the island
+        Get credentials from the island
 
         :raises IslandAPIConnectionError: If the client could not connect to the island
         :raises IslandAPIRequestError: If there was a problem with the client request
         :raises IslandAPIRequestFailedError: If the server experienced an error
         :raises IslandAPITimeoutError: If the command timed out
-        :return: Agent configuration
+        :return: Credentials
         """
 
     @abstractmethod
-    def get_credentials_for_propagation(self) -> Sequence[Credentials]:
+    def get_agent_signals(self, agent_id: str) -> Optional[datetime]:
         """
-        Get credentials from the island
+        Get agent signals from the island
 
         :raises IslandAPIConnectionError: If the client could not connect to the island
         :raises IslandAPIRequestError: If there was a problem with the client request
         :raises IslandAPIRequestFailedError: If the server experienced an error
         :raises IslandAPITimeoutError: If the command timed out
-        :return: Credentials
+        :return: Terminate datetime
```

Comment:
Shouldn't mention data types in docstrings
```suggestion
        :return: Terminate signal's timestamp
```